task_02/tevfik_aksoy/main.cpp

```diff
@@ -0,0 +1,183 @@
+#include <iostream>
+#include <vector>
+#include <string>
+#include <tuple>
+#include <set>
+#include <fstream>
+
+using std::cout;
+using std::endl;
+using std::get;
+using std::to_string;
+using std::vector;
+
+//Tuple of <int,int> for realization of vector of vectors of <int,int>
+typedef std::tuple<int, int> DestID_EdgeID;
+
+//Structure of nodes(vertices)
+struct Node
+{
+private:
+	int id = 0;
+
+public:
+	explicit Node(int id) : id(id) {}
+	explicit Node() : id(0) {}
+
+	const int GetID() const
+	{
+		return id;
+	}
+
+	void SetID(int val)
+	{
+		id = val;
+	}
+
+	std::string to_json(vector<int> edge_ids, bool end_flag) const
+	{
+		std::string str = "\t{ \"id\": " + to_string(id) + ", \"edge_ids\": [";
+		for (int i = 0; i < edge_ids.size(); i++)
+		{
+			str += to_string(edge_ids[i]);
+			if (i + 1 == edge_ids.size())
+			{
+				str += "] }";
+				if (!end_flag)
+					str += ",\n";
+				else
+					str += "\n  ],\n";
+			}
+			else
+				str += ", ";
+		}
+		return str;
+	}
+};
+
+//Structure of the edge including the source and destination nodes in it
+struct Edge
+{
+private:
+	int id = 0;
+
+public:
+	Node src_node;
+	Node dest_node;
+	Edge(int src_id, int dest_id, int id)
+	{
+		src_node.SetID(src_id);
+		dest_node.SetID(dest_id);
+		this->id = id;

```

Comment:
Чем обращаться к `this` лучше просто переименуй локальную. переменную, например в `_id`
```suggestion
              id = _id;
```